### PR TITLE
first ver.

### PR DESCRIPTION
**Pull request by vitalygv**
Hello, please check this out

### CODE REVIEW

File: tic-tac-toe.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    switch_players
+    puts "Make your move player #{@current_player}"
+    player_move = gets.chomp
+    while @board[player_move.to_i - 1] == 'x' || @board[player_move.to_i - 1] == 'o'
+      puts 'NO'
+      puts "Make your move again player #{@current_player}"
+      player_move = gets.chomp
+    end
+    put_sign(player_move.to_i, @current_sign)
+    @current_moves.push(player_move.to_i)
```
Comment (krabique) [inline tic-tac-toe.rb:54]: there are a lot of places with `player_move.to_i`, is it expected to be a non-integer?

File: tic-tac-toe.rb [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+  def move
+    switch_players
+    puts "Make your move player #{@current_player}"
+    player_move = gets.chomp
+    while @board[player_move.to_i - 1] == 'x' || @board[player_move.to_i - 1] == 'o'
+      puts 'NO'
+      puts "Make your move again player #{@current_player}"
+      player_move = gets.chomp
+    end
```
Comment (krabique) [inline tic-tac-toe.rb:52]: I'd rather move this to a new method, and refactor it.

Use `loop`

```ruby
loop do
  player_move = gets.chomp
  break player_move if ["x", "o"].include?(@board[player_move.to_i - 1])
  
  puts 'NO'
  puts "Make your move again player #{@current_player}"
end
```

File: tic-tac-toe.rb [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    @current_player = 1
+    @current_sign = 'x'
+    @x_moves = []
+    @o_moves = []
+    @current_moves = []
+  end
+
+  def show_board
+    screen = @board.join(' ')
+    (0..@board.length - 1).each.with_index do |i, index|
```
Comment (krabique) [inline tic-tac-toe.rb:18]: So what is the difference between i and index?

Use `@board.each.with_index do |_board_cell,  index|` or something like that

File: tic-tac-toe.rb [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    @turn = 1
+    @current_player = 1
+    @current_sign = 'x'
+    @x_moves = []
+    @o_moves = []
+    @current_moves = []
+  end
+
+  def show_board
+    screen = @board.join(' ')
```
Comment (krabique) [inline tic-tac-toe.rb:17]: It's not used anywhere

File: tic-tac-toe.rb [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    (0..@board.length - 1).each.with_index do |i, index|
+      if index == 2 || index == 5
```
Comment (krabique) [inline tic-tac-toe.rb:19]: magic numbers, also it's better to use `any?` or `includes?`

File: tic-tac-toe.rb [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        puts @board[i]
+      else print @board[i]
```
[the remark this answers is not in the document]
Comment (vitalygv) [reply tic-tac-toe.rb:21]: I used replit autoformat so probably it gave weird indentations at some lines

File: tic-tac-toe.rb (hunk 1 above)
Comment (vitalygv) [reply tic-tac-toe.rb:54]: It gives an error if there is no `to_i` conversion for some reason. So I changed it like this `player_move = gets.chomp.to_i`

File: tic-tac-toe.rb (hunk 4 above)
Comment (vitalygv) [reply tic-tac-toe.rb:17]: removed

File: tic-tac-toe.rb (hunk 2 above)
Comment (vitalygv) [reply tic-tac-toe.rb:52]:
```
` while @board[player_move - 1].class == String
      puts 'NO'
      puts "Make your move again player #{@current_player}"
      player_move = gets.chomp.to_i
    end`
```

Comment (vitalygv) [reply tic-tac-toe.rb:52]: what if I just check if this cell is of String type which mean it has "x" or "o" instead of cell number?

**Pull request merged**